src/main/scala/cognite/spark/v1/AlphaDataModelInstancesHelper.scala

```diff
@@ -114,34 +114,39 @@ object AlphaDataModelInstancesHelper {
       case a => throw new CdfSparkException(unknownPropertyTypeMessage(a))
     }
 
-  private def toDateProperty: Any => DataModelProperty[_] = {
-    case x: LocalDate => PropertyType.Date.Property(x)
-    case x: java.sql.Date => PropertyType.Date.Property(x.toLocalDate)
-    case x: LocalDateTime => PropertyType.Date.Property(x.toLocalDate)
+  private def toDateProperty: Any => Option[DataModelProperty[_]] = {
+    case x: LocalDate => Some(PropertyType.Date.Property(x))
+    case x: java.sql.Date => Some(PropertyType.Date.Property(x.toLocalDate))
+    case x: LocalDateTime => Some(PropertyType.Date.Property(x.toLocalDate))
     case a => throw new CdfSparkException(notValidPropertyTypeMessage(a, "date", Some("date")))
   }
 
-  private def toTimestampProperty: Any => DataModelProperty[_] = {
+  private def toTimestampProperty: Any => Option[DataModelProperty[_]] = {
     case x: Instant =>
-      PropertyType.Timestamp.Property(OffsetDateTime.ofInstant(x, ZoneId.of("UTC")).toZonedDateTime)
+      Some(
+        PropertyType.Timestamp.Property(OffsetDateTime.ofInstant(x, ZoneId.of("UTC")).toZonedDateTime))
     case x: java.sql.Timestamp =>
       // Using ZonedDateTime directly without OffSetDateTime, adds ZoneId to the end of encoded string
       // 2019-10-02T07:00+09:00[Asia/Seoul] instead of 2019-10-02T07:00+09:00, API does not like it.
-      PropertyType.Timestamp.Property(
-        OffsetDateTime.ofInstant(x.toInstant, ZoneId.of("UTC")).toZonedDateTime)
+      Some(
+        PropertyType.Timestamp.Property(
+          OffsetDateTime.ofInstant(x.toInstant, ZoneId.of("UTC")).toZonedDateTime))
     case x: ZonedDateTime =>
-      PropertyType.Timestamp.Property(x)
+      Some(PropertyType.Timestamp.Property(x))
     case a =>
       throw new CdfSparkException(
         notValidPropertyTypeMessage(a, PropertyType.Timestamp.code, Some("timestamp")))
   }
 
-  private def toDirectRelationProperty: Any => DataModelProperty[_] = {
+  private def toDirectRelationProperty: Any => Option[DataModelProperty[_]] = {
```

Comment:
@vuhainguyen so this makes `array("something", null)` effectless, meaning it writes null to that property. and the cases other than `array("space", "ext_id")` will fail.

Reply:
need to test though, but this should solve what you described.

Reply:
ok